src/utils/device.js

```diff
@@ -146,13 +146,24 @@ function isIpad (mockUserAgent, mockDevicePlatform, mockDeviceTouchPoints) {
 }
 module.exports.isIpad = isIpad;
 
+/**
+ *  Detect Apple Vision Pro devices.
+ *  Safari for Apple Vision Pro presents itself as a desktop browser.
+*/
+function isAppleVisionPro () {
+  var isMacintosh = navigator.userAgent.includes('Macintosh');
+  var hasFiveTouchPoints = navigator.maxTouchPoints === 5;
+  return isMacintosh && hasFiveTouchPoints;
+}
+module.exports.isAppleVisionPro = isAppleVisionPro;
+
 function isIOS () {
   return /iPad|iPhone|iPod/.test(window.navigator.platform);
 }
 module.exports.isIOS = isIOS;
 
 function isMobileDeviceRequestingDesktopSite () {
-  return !isMobile() && !isMobileVR() && window.orientation !== undefined;
+  return !isMobile() && !isAppleVisionPro() && !isMobileVR() && window.orientation !== undefined;
```

Comment:
The condition should be part of `isMobileVR()`.  That should also help with https://github.com/aframevr/aframe/issues/5530

Reply:
thanks. done